src/expr/interpret/eval.hpp

```diff
@@ -100,6 +100,28 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
 #undef BINARY_OPERATOR_VISITOR
 #undef UNARY_OPERATOR_VISITOR
 
+  void HandleShardError(Error &shard_error, const std::string_view accessed_object) {
```

Comment:
I think this name is very generic. Maybe consider making it more specific to read operations.